-DNO_METAL=ON breaks build on Mac OS for 3.5.0

Compiling OpenSubdiv without Metal breaks the build for the 3.5.0 release.

It seems that the new bfr addition requires `-std=11`, which is only added if Metal support is enabled (see [CMakeLists.txt:444](https://github.com/PixarAnimationStudios/OpenSubdiv/blob/release/CMakeLists.txt#L444)).

Tested with 3.5.0 Release on Apple clang version 13.1.6

Steps to reproduce:
```shell
mkdir build && cd build
cmake -DNO_METAL=ON ..
make
```

make output:

```
[  1%] Building CXX object glLoader/CMakeFiles/glLoader_obj.dir/glLoader.cpp.o
[  1%] Building CXX object glLoader/CMakeFiles/glLoader_obj.dir/glApi.cpp.o
[  1%] Built target glLoader_obj
[  2%] Building CXX object opensubdiv/osd/CMakeFiles/osd_cpu_obj.dir/cpuEvaluator.cpp.o
[  2%] Building CXX object opensubdiv/osd/CMakeFiles/osd_cpu_obj.dir/cpuKernel.cpp.o
[  3%] Building CXX object opensubdiv/osd/CMakeFiles/osd_cpu_obj.dir/cpuPatchTable.cpp.o
[  4%] Building CXX object opensubdiv/osd/CMakeFiles/osd_cpu_obj.dir/cpuVertexBuffer.cpp.o
[  4%] Built target osd_cpu_obj
[  4%] Building CXX object opensubdiv/sdc/CMakeFiles/sdc_obj.dir/crease.cpp.o
[  5%] Building CXX object opensubdiv/sdc/CMakeFiles/sdc_obj.dir/typeTraits.cpp.o
[  5%] Built target sdc_obj
[  6%] Building CXX object opensubdiv/vtr/CMakeFiles/vtr_obj.dir/fvarLevel.cpp.o
[  6%] Building CXX object opensubdiv/vtr/CMakeFiles/vtr_obj.dir/fvarRefinement.cpp.o
[  7%] Building CXX object opensubdiv/vtr/CMakeFiles/vtr_obj.dir/level.cpp.o
[  8%] Building CXX object opensubdiv/vtr/CMakeFiles/vtr_obj.dir/quadRefinement.cpp.o
[  8%] Building CXX object opensubdiv/vtr/CMakeFiles/vtr_obj.dir/refinement.cpp.o
[  9%] Building CXX object opensubdiv/vtr/CMakeFiles/vtr_obj.dir/sparseSelector.cpp.o
[ 10%] Building CXX object opensubdiv/vtr/CMakeFiles/vtr_obj.dir/triRefinement.cpp.o
[ 10%] Built target vtr_obj
[ 11%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/bilinearPatchBuilder.cpp.o
[ 12%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/catmarkPatchBuilder.cpp.o
[ 12%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/error.cpp.o
[ 13%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/loopPatchBuilder.cpp.o
[ 14%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/patchBasis.cpp.o
[ 14%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/patchBuilder.cpp.o
[ 15%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/patchDescriptor.cpp.o
[ 16%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/patchMap.cpp.o
[ 16%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/patchTable.cpp.o
[ 17%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/patchTableFactory.cpp.o
[ 18%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/ptexIndices.cpp.o
[ 18%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/stencilTable.cpp.o
[ 19%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/stencilTableFactory.cpp.o
[ 20%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/stencilBuilder.cpp.o
[ 20%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/topologyDescriptor.cpp.o
[ 21%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/topologyRefiner.cpp.o
[ 22%] Building CXX object opensubdiv/far/CMakeFiles/far_obj.dir/topologyRefinerFactory.cpp.o
[ 22%] Built target far_obj
[ 23%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/faceSurface.cpp.o
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/faceSurface.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceSurface.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceTopology.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceVertex.h:33:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceData.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:86:50: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization(Parameterization const &) = default;
                                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:87:62: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization & operator=(Parameterization const &) = default;
                                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:88:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Parameterization() = default;
                          ^
3 warnings generated.
[ 23%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/faceTopology.cpp.o
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/faceTopology.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceTopology.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceVertex.h:33:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceData.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:86:50: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization(Parameterization const &) = default;
                                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:87:62: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization & operator=(Parameterization const &) = default;
                                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:88:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Parameterization() = default;
                          ^
3 warnings generated.
[ 24%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/faceVertex.cpp.o
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/faceVertex.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceVertex.h:33:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceData.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:86:50: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization(Parameterization const &) = default;
                                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:87:62: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization & operator=(Parameterization const &) = default;
                                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:88:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Parameterization() = default;
                          ^
3 warnings generated.
[ 25%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/hash.cpp.o
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/hash.cpp:319:12: warning: private field 'm_data' is not used [-Wunused-private-field]
    uint64 m_data[2*sc_numVars];   // unhashed data, for partial messages
           ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/hash.cpp:320:12: warning: private field 'm_state' is not used [-Wunused-private-field]
    uint64 m_state[sc_numVars];  // internal state of the hash
           ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/hash.cpp:321:12: warning: private field 'm_length' is not used [-Wunused-private-field]
    size_t m_length;             // total length of the input so far
           ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/hash.cpp:322:12: warning: private field 'm_remainder' is not used [-Wunused-private-field]
    uint8  m_remainder;          // length of unhashed data stashed in m_data
           ^
4 warnings generated.
[ 25%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/irregularPatchBuilder.cpp.o
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/irregularPatchBuilder.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/irregularPatchBuilder.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceSurface.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceTopology.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceVertex.h:33:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceData.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:86:50: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization(Parameterization const &) = default;
                                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:87:62: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization & operator=(Parameterization const &) = default;
                                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:88:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Parameterization() = default;
                          ^
3 warnings generated.
[ 26%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/parameterization.cpp.o
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/parameterization.cpp:25:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:86:50: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization(Parameterization const &) = default;
                                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:87:62: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization & operator=(Parameterization const &) = default;
                                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:88:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Parameterization() = default;
                          ^
3 warnings generated.
[ 27%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/patchTree.cpp.o
[ 27%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/patchTreeBuilder.cpp.o
[ 28%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/refinerSurfaceFactory.cpp.o
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/refinerSurfaceFactory.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactory.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surface.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceData.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:86:50: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization(Parameterization const &) = default;
                                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:87:62: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization & operator=(Parameterization const &) = default;
                                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:88:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Parameterization() = default;
                          ^
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/refinerSurfaceFactory.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactory.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surface.h:94:36: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Surface(Surface const & src) = default;
                                   ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surface.h:95:47: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Surface& operator=(Surface const & src) = default;
                                              ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surface.h:96:18: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Surface() = default;
                 ^
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/refinerSurfaceFactory.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactory.h:161:23: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
    ~SurfaceFactory() override;
                      ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactory.h:324:24: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <typename REAL=float>
                       ^    ~~~~~
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactory.h:328:24: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <typename REAL=float>
                       ^    ~~~~~
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactory.h:332:24: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <typename REAL=float>
                       ^    ~~~~~
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactory.h:336:24: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <typename REAL=float>
                       ^    ~~~~~
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactory.h:365:46: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    SurfaceFactory(SurfaceFactory const &) = delete;
                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactory.h:366:58: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    SurfaceFactory & operator=(SurfaceFactory const &) = delete;
                                                         ^
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/refinerSurfaceFactory.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:31:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactoryCache.h:64:56: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    SurfaceFactoryCache(SurfaceFactoryCache const &) = delete;
                                                       ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactoryCache.h:65:68: warning: deleted function definitions are a C++11 extension [-Wc++11-extensions]
    SurfaceFactoryCache & operator=(SurfaceFactoryCache const &) = delete;
                                                                   ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactoryCache.h:121:36: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
    ~SurfaceFactoryCacheThreaded() override = default;
                                   ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactoryCache.h:121:47: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~SurfaceFactoryCacheThreaded() override = default;
                                              ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactoryCache.h:128:46: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
    DataType Find(KeyType const & key) const override {
                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceFactoryCache.h:133:62: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
    DataType Add(KeyType const & key, DataType const & data) override {
                                                             ^
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/refinerSurfaceFactory.cpp:25:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:69:34: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
    ~RefinerSurfaceFactoryBase() override = default;
                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:69:45: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~RefinerSurfaceFactoryBase() override = default;
                                            ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:93:45: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
    bool isFaceHole( Index faceIndex) const override;
                                            ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:94:45: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
    int  getFaceSize(Index faceIndex) const override;
                                            ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:97:54: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
                        Index vertexIndices[]) const override;
                                                     ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:99:72: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
                        FVarID fvarID, Index fvarValueIndices[]) const override;
                                                                       ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:102:68: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
                        VertexDescriptor * vertexDescriptor) const override;
                                                                   ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:106:54: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
                        Index vertexIndices[]) const override;
                                                     ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:109:72: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
                        FVarID fvarID, Index fvarValueIndices[]) const override;
                                                                       ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:114:54: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
                        Index vertexIndices[]) const override;
                                                     ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:118:72: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
                        FVarID fvarID, Index fvarValueIndices[]) const override;
                                                                       ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:166:30: warning: 'override' keyword is a C++11 extension [-Wc++11-extensions]
    ~RefinerSurfaceFactory() override = default;
                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/refinerSurfaceFactory.h:166:41: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~RefinerSurfaceFactory() override = default;
                                        ^
32 warnings generated.
[ 29%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/regularPatchBuilder.cpp.o
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/regularPatchBuilder.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/regularPatchBuilder.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceSurface.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceTopology.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/faceVertex.h:33:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceData.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:86:50: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization(Parameterization const &) = default;
                                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:87:62: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization & operator=(Parameterization const &) = default;
                                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:88:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Parameterization() = default;
                          ^
3 warnings generated.
[ 29%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/surface.cpp.o
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/surface.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surface.h:30:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceData.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:86:50: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization(Parameterization const &) = default;
                                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:87:62: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization & operator=(Parameterization const &) = default;
                                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:88:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Parameterization() = default;
                          ^
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/surface.cpp:25:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surface.h:94:36: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Surface(Surface const & src) = default;
                                   ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surface.h:95:47: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Surface& operator=(Surface const & src) = default;
                                              ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surface.h:96:18: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Surface() = default;
                 ^
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/surface.cpp:27:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/pointOperations.h:230:19: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <int SIZE = 0>
                  ^      ~
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/pointOperations.h:281:19: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <int SIZE = 0>
                  ^      ~
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/pointOperations.h:332:19: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <int SIZE = 0>
                  ^      ~
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/pointOperations.h:399:19: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <int SIZE = 0>
                  ^      ~
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/pointOperations.h:453:19: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <int SIZE = 0>
                  ^      ~
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/pointOperations.h:510:19: warning: default template arguments for a function template are a C++11 extension [-Wc++11-extensions]
    template <int SIZE = 0>
                  ^      ~
12 warnings generated.
[ 30%] Building CXX object opensubdiv/bfr/CMakeFiles/bfr_obj.dir/surfaceData.cpp.o
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/surfaceData.cpp:25:
In file included from /Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/surfaceData.h:30:
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:86:50: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization(Parameterization const &) = default;
                                                 ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:87:62: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    Parameterization & operator=(Parameterization const &) = default;
                                                             ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/../bfr/parameterization.h:88:27: warning: defaulted function definitions are a C++11 extension [-Wc++11-extensions]
    ~Parameterization() = default;
                          ^
/Users/jan/workspace/OpenSubdiv/opensubdiv/bfr/surfaceData.cpp:76:17: error: no viable overloaded '='
    _irregPatch = 0;
    ~~~~~~~~~~~ ^ ~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:511:17: note: candidate function not viable: no known conversion from 'int' to 'const std::shared_ptr<const OpenSubdiv::v3_5_0::Bfr::PatchTree>' for 1st argument
    shared_ptr& operator=(const shared_ptr& __r) _NOEXCEPT;
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:519:9: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-0>' against 'int'
        operator=(const shared_ptr<_Yp>& __r) _NOEXCEPT;
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:521:17: note: candidate function not viable: no known conversion from 'int' to 'std::shared_ptr<const OpenSubdiv::v3_5_0::Bfr::PatchTree>' for 1st argument
    shared_ptr& operator=(shared_ptr&& __r) _NOEXCEPT;
                ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:529:9: note: candidate template ignored: could not match 'shared_ptr<type-parameter-0-0>' against 'int'
        operator=(shared_ptr<_Yp>&& __r);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:539:9: note: candidate template ignored: could not match 'auto_ptr<type-parameter-0-0>' against 'int'
        operator=(auto_ptr<_Yp>&& __r);
        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk/usr/include/c++/v1/__memory/shared_ptr.h:548:9: note: candidate template ignored: could not match 'unique_ptr<type-parameter-0-0, type-parameter-0-1>' against 'int'
        operator=(unique_ptr<_Yp, _Dp>&& __r);
        ^
3 warnings and 1 error generated.
make[2]: *** [opensubdiv/bfr/CMakeFiles/bfr_obj.dir/surfaceData.cpp.o] Error 1
make[1]: *** [opensubdiv/bfr/CMakeFiles/bfr_obj.dir/all] Error 2
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,17 @@ cmake_minimum_required(VERSION 3.12)
 
 project(OpenSubdiv)
 
+# Set C++ standard requirements, allowing overrides
+if (NOT DEFINED CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 14)
+endif()
+if (NOT DEFINED CMAKE_CXX_STANDARD_REQUIRED)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+endif()
+if (NOT DEFINED CMAKE_CXX_EXTENSIONS)
+    set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+
 # Turn on folder support
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
@@ -437,12 +448,6 @@ endif()
 
 if( METAL_FOUND AND NOT NO_METAL)
     set(OSD_GPU TRUE)
-    if (CMAKE_COMPILER_IS_CLANGCC)
-        # When building with Metal support enabled, we need to explicitly
-        # request that the C++ code be compiled with C++11 since some example
-        # code for Metal relies on functionality only available in C++11.
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-    endif()
 endif()
 
 if( OPENGL_FOUND AND NOT NO_OPENGL)

--- a/examples/common/CMakeLists.txt
+++ b/examples/common/CMakeLists.txt
@@ -157,8 +157,6 @@ if(METAL_FOUND)
         mtlHud.h
     )
 
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
-
     add_library(examples_common_mtl_obj
         OBJECT
             ${EXAMPLES_COMMON_SOURCE_FILES}

--- a/examples/mtlPtexViewer/CMakeLists.txt
+++ b/examples/mtlPtexViewer/CMakeLists.txt
@@ -108,10 +108,6 @@ if(IOS)
     )
 endif()
 
-
-
-set_property (TARGET mtlPtexViewer PROPERTY CXX_STANDARD 14)
-
 target_link_libraries(mtlPtexViewer
     ${PLATFORM_LIBRARIES}
 )

--- a/examples/mtlViewer/CMakeLists.txt
+++ b/examples/mtlViewer/CMakeLists.txt
@@ -104,10 +104,6 @@ if(IOS)
     )
 endif()
 
-
-
-set_property (TARGET mtlViewer PROPERTY CXX_STANDARD 14)
-
 target_link_libraries(mtlViewer
     ${PLATFORM_LIBRARIES}
 )

--- a/opensubdiv/CMakeLists.txt
+++ b/opensubdiv/CMakeLists.txt
@@ -80,7 +80,6 @@ if (NOT NO_LIB)
         list(APPEND PLATFORM_GPU_LIBRARIES
             ${METAL_LIBRARIES}
         )
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
     endif()
 
     if ( OPENCL_FOUND )

